core/shared/src/main/scala/scalaz/zio/Queue.scala

```diff
@@ -1,158 +1,408 @@
-// Copyright (C) 2018 John A. De Goes. All rights reserved.
-
 package scalaz.zio
 
-import scala.collection.immutable.{ Queue => IQueue }
+// Copyright (C) 2018 John A. De Goes. All rights reserved.
 
-import internal._
+import scala.collection.immutable.{ Queue => IQueue }
+import Queue.internal._
 
 /**
- * A `Queue` is a lightweight, asynchronous queue. This implementation is
- * naive, if functional, and could benefit from significant optimization.
- *
- * TODO:
- *
- * 1. Investigate using a faster option than `Queue`, because `Queue` has
- *    `O(n)` `length` method.
- * 2. Benchmark to see how slow this implementation is and if there are any
- *    easy ways to improve performance.
- */
-class Queue[A] private (capacity: Int, ref: Ref[State[A]]) extends Serializable {
+ * A `Queue` is a lightweight, asynchronous queue. This implementation is
+ * naive, if functional, and could benefit from significant optimization.
+ *
+ * TODO:
+ *
+ * 1. Investigate using a faster option than `Queue`, because `Queue` has
+ *    `O(n)` `length` method.
+ * 2. Benchmark to see how slow this implementation is and if there are any
+ *    easy ways to improve performance.
+ */
+class Queue[A] private (
```

Comment:
Queue should ` extends Serializable`  then check the `SerializableSpec`

Reply:
Thanks, @wi101 !